src/stan/io/reader.hpp

```diff
@@ -995,8 +995,10 @@ namespace stan {
        * Cholesky factor for a correlation matrix.
        */
       inline matrix_t cholesky_corr(size_t K) {
+        using stan::error_handling::check_cholesky_factor_corr;
         matrix_t y(matrix(K,K));
```

Comment:
`matrix` is actually a method defined in the reader that creates a matrix by reading `K * K` elements from the reader and building a matrix from them. The existing code is correct. If anything, we should rename the `matrix` method.